src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -190,6 +209,62 @@ trait AttributeComponent {
       filter(_.id inSetBind attributeRecords.map(_.id)).delete
     }
 
+    object AlterAttributesUsingTempTableQueries extends RawSqlQuery {
+      val driver: JdbcDriver = AttributeComponent.this.driver
+      /*!CAUTION! - do not drop the ownerIds clause at the end or it will delete every attribute in the table!
+
+        deleteMasterFromAction: deletes any row in *_ATTRIBUTE that doesn't
+          exist in *_ATTRIBUTE_TEMP but exists in *_ATTRIBUTE (bound by ownerIds)
+        updateInMasterAction: updates any row in *_ATTRIBUTE that also
+          exists in *_ATTRIBUTE_TEMP (bound by ownerIds)
+        insertIntoMasterAction: inserts any row into *_ATTRIBUTE that doesn't
+          exist in *_ATTRIBUTE but exists in *_ATTRIBUTE_TEMP (bound by ownerIds)
+
+        These left joins give us the rows related to the attributes that are in the temp table.
+        We filter on ta.owner_id == null if we want to see that the row does NOT exist in the temp table
+        We filter on ea.owner_id == null if we want to see that the row does NOT exist in the real table
+        We filter on ta.owner_id != null if we want to see that the row exists in both of the tables
+       */
+      def deleteFromMasterAction(ownerIds: String) =
+        sql"""delete ea from #${baseTableRow.tableName} ea
+                left join #${baseTableRow.tableName}_TEMP ta
+                on (ea.name,ea.owner_id,ea.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
+                where ta.owner_id is null and ea.owner_id in #$ownerIds""".as[Int]
+      def updateInMasterAction(ownerIds: String) =
+        sql"""update #${baseTableRow.tableName} ea
+                left join #${baseTableRow.tableName}_TEMP ta
+                on (ea.name,ea.owner_id,ea.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
+                set ea.value_string=ta.value_string, ea.value_number=ta.value_number, ea.value_boolean=ta.value_boolean, ea.value_entity_ref=ta.value_entity_ref
+                where ta.owner_id is not null and ea.owner_id in #$ownerIds""".as[Int]
+      def insertIntoMasterAction(ownerIds: String) =
+        sql"""insert into #${baseTableRow.tableName}(name,value_string,value_number,value_boolean,value_entity_ref,list_index,owner_id)
+                select ta.name,ta.value_string,ta.value_number,ta.value_boolean,ta.value_entity_ref,ta.list_index,ta.owner_id
+                from #${baseTableRow.tableName}_TEMP ta
+                left join #${baseTableRow.tableName} ea
+                on (ea.name,ea.owner_id,ea.list_index)<=>(ta.name,ta.owner_id,ta.list_index)
+                where ea.owner_id is null and ta.owner_id in #${ownerIds}""".as[Int]
+
+      def createAttributeTempTableAction(ownerId: OWNER_ID) = {
```

Comment:
i'm assuming there is a better way to do a match on the OWNER_ID type without having to actually pass through an ownerId?